frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,499 @@
+      },
+    };
+  }
+
+  componentDidMount() {
+    const { kind, metadata } = this.state.resourceObj;
+    const ko = _.find(k8sModels, model => model.plural === kind);
+    const supportedKinds = [
+      'Deployment',
+      'DeploymentConfig',
+      'ReplicaSet',
+      'ReplicationController',
+    ];
+
+    if (!_.includes(supportedKinds, ko.kind)) {
+      this.setState({
+        error: `Storage is not supported for kind ${ko.kind}.`,
+      });
+      return;
+    }
+
+    // Get the current resource so we can add to its definition
+    k8sGet(ko, metadata.name, metadata.namespace).then(
+      resourceObj => {
+        this.setState({ resourceObj });
+      },
+      err => this.setState({ error: err.message })
+    );
+  }
+
+  onError = error => {
+    this.setState({
+      error,
+      inProgress: false,
+    });
+  };
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    const { name, value } = event.currentTarget;
+    const formData = { ...this.state.formData, [name]: value };
+    this.setState({ formData });
+  };
+
+  handlePvcChange = event => {
+    const { name, value } = event.target;
+    const formData = { ...this.state.formData, [name]: value };
+    const storage = updateVolumeName(
+      this.state.resourceObj,
+      this.state.storage,
+      value
+    );
+    formData.volumeName = storage.volumeName;
+    this.setState({ formData, storage });
+  };
+
+  onMountAsReadOnlyChanged: React.ReactEventHandler<
+    HTMLInputElement
+  > = event => {
+    const mountAsReadOnly = !this.state.mountAsReadOnly;
+    this.setState({ mountAsReadOnly });
+    // Now deal with selecting checkbox
+    let value = event.currentTarget.value;
+    let checked = new Set(this.state.selectedOptions);
+    if (checked.has(value)) {
+      checked.delete(value);
+    } else {
+      checked.add(value);
+    }
+    this.setState({ selectedOptions: Array.from(checked) });
+  };
+
+  onFormDisable = disable => {
+    this.setState({
+      disableForm: disable,
+    });
+  };
+
+  validateForm = () => {
+    // Check that a valid PVC has been choosen
+    if (!this.state.formData.claimName) {
+      this.setState({ error: 'Storage must be selected' });
+      return false;
+    }
+    return true;
+  };
+
+  // Add Attach storage //
+  save = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    if (!this.validateForm()) {
+      return;
+    }
+    const { kind, metadata } = this.state.resourceObj;
+    const ko = _.find(k8sModels, model => model.kind === kind);
+    const resourceObj = { ...this.state.resourceObj };
+    const template = resourceObj.spec.template;
+    const { mountPath, subPath, claimName } = this.state.formData;
+    let { volumeName } = this.state.formData;
+    const mountAsReadOnly = this.state.mountAsReadOnly;
+    const { storage } = this.state;
+    const isContainerSelected = ({ name }) =>
+      storage.allContainers || storage.containers[name];
```

Comment:
We have the logic here for selecting a container, but I don't see the form element for actually picking the container?

We should add this eventually. You don't always want to mount the same volume at the same path in all containers.

Reply:
agreed.